src/util/compat.h

```diff
@@ -15,6 +15,11 @@
 #define _POSIX_C_SOURCE 200809L
 #endif
 
+// Apple doing weird stuff again
```

Comment:
This is probably worth briefly explaining, e.g. "_POSIX_C_SOURCE makes some standard definitions disappear from Apple's headers unless this is defined". Expletives permitted.